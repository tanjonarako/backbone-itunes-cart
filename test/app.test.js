import Backbone from "backbone";
import AppView from "../src/app";

describe('views:AppView', function() {
    it('should render a list', function() {
        const appView = new AppView().render();

        expect(appView.$el.html()).toMatchSnapshot();
    });

    it('should add a song into cart', function() {
        const appView = new AppView().render();

        appView.searchCollection.add({trackId: '1', trackName: 'Shape of you', artistName: 'Ed Sheeran'});
        appView.searchCollection.add({trackId: '2', trackName: 'Perfect', artistName: 'Ed Sheeran'});

        appView._addToCart('1');

        expect(appView.cartCollection.length).toEqual(1);
    });

    it('should remove a song into cart', function() {
        const appView = new AppView().render();

        appView.searchCollection.add({trackId: '1', trackName: 'Shape of you', artistName: 'Ed Sheeran'});
        appView.cartCollection.add({trackId: '1', trackName: 'Shape of you', artistName: 'Ed Sheeran'});

        appView._removeFromCart('1');

        expect(appView.cartCollection.length).toEqual(0);
    });
});
