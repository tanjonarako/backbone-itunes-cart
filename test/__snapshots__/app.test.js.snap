// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views:AppView should render a list 1`] = `
"<div class=\\"container-fluid text-center\\">
    <div class=\\"row justify-content-center align-items-center search-bar my-2\\"><form class=\\"form-inline\\">
    <div class=\\"form-group\\">
        <input class=\\"inputSearch form-control mr-1\\" type=\\"text\\" placeholder=\\"Search\\">
        <button type=\\"submit\\" class=\\"btn btn-secondary cursor-pointer\\"><i class=\\"fa fa-search\\" aria-hidden=\\"true\\"></i>Search</button>
    </div>
</form>
</div>
    <div class=\\"row\\">
        <div class=\\"search-result col\\">
</div>
        <div class=\\"cart col\\">
</div>
    </div>
</div>
"
`;
